languages/go/exercises/concept/booleans/booleans_test.go

```diff
@@ -0,0 +1,202 @@
+package booleans
+
+import "testing"
+
+func TestCanFastAttack(t *testing.T) {
+	tests := []struct {
+		name          string
+		knightIsAwake bool
+		expected      bool
+	}{
+		{
+			name:          "Knight is awake",
+			knightIsAwake: true,
+			expected:      false,
+		},
+		{
+			name:          "Knight is sleeping",
+			knightIsAwake: false,
+			expected:      true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := CanFastAttack(tt.knightIsAwake); got != tt.expected {
+				t.Errorf("CanFastAttack(%v) = %v; want %v", tt.knightIsAwake, tt.knightIsAwake, tt.expected)
+			}
+		})
+	}
+}
+
+func TestCanSpy(t *testing.T) {
+	tests := []struct {
+		desc                      string
+		characterStateCombination []bool
+		expected                  bool
+	}{
+		{
+			desc:                      "All characters are sleeping",
+			characterStateCombination: []bool{false, false, false},
```

Comment:
I'd probably have 3 variables for this with names so it is more obvious what is what. Even in the tests as students might solve this locally and look at the tests.

```suggestion
			knightIsAwake:    false,
			archerIsAwake:    false,
			prisonerIsAwake: false,
```

Reply:
Sure. It's clearer this way.